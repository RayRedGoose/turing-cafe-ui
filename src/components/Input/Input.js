import React from 'react'
import './Input.scss'

const Input = ({type, action, currentValue}) => {

  const makeChanges = (event) => {
    action(type, event.target.value)
  }

  const placeholder = (type === 'number') ? 'number of guests' : type
  return (
    <input
      className="reserv-input"
      type="text"
      name={type}
      value={currentValue}
      placeholder={placeholder}
      onChange={makeChanges}/>
  )
}

export default Input
