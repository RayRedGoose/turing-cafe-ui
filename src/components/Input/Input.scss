.reserv-input {
  font-size: 16px;
  text-transform: capitalize;
  padding: 0.4em;
  margin-right: 20px;
}
