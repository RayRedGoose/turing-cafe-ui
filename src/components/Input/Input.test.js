import React from 'react'
import { shallow } from 'enzyme'
import Input from './Input'

describe("Input", () => {
  let input
  const action = jest.fn()

  beforeEach(() => {
    input = shallow(
      <Input
      key='name'
      type='name'
      currentValue='Ray'
      action={action} />
    )
  })

  it('renders with all data passed in correctlly', () => {
    expect(input).toMatchSnapshot()
  })
  
  it('should call action prop when changes are happening', () => {
    input.find('input').simulate(
      'change',
      {target: {value: 'Mark'}}
    )

    expect(action).toHaveBeenCalled()
  })
})
