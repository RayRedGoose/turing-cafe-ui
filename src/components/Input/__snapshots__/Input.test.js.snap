// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders with all data passed in correctlly 1`] = `
<input
  className="reserv-input"
  name="name"
  onChange={[Function]}
  placeholder="name"
  type="text"
  value="Ray"
/>
`;
